modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java

```diff
@@ -799,7 +799,19 @@ public ANode visitNull(NullContext ctx) {
 
     @Override
     public ANode visitString(StringContext ctx) {
-        String string = ctx.STRING().getText().substring(1, ctx.STRING().getText().length() - 1);
+        String string = ctx.STRING().getText();
+        boolean doubleQuoted = string.charAt(0) == '"';
+
+        // Strip the leading and trailing quotes
+        string = string.substring(1, ctx.STRING().getText().length() - 1);
```

Comment:
Can we do all of these things (stripping last char, backslash escape and quote escape) in one pass with a string builder? Right 3 copies are made of the string.